src/module.ts

```diff
@@ -0,0 +1,41 @@
+import { defineNuxtModule, createResolver, addImportsDir, addComponentsDir, installModule } from '@nuxt/kit'
+import { fileURLToPath } from 'url'
+import { defu } from 'defu'
+
+export type ModuleOptions = {
+  cloudName?: string;
+}
+
+export default defineNuxtModule<ModuleOptions>({
+  meta: {
+    name: '@nuxtjs/cloudinary',
+    configKey: 'cloudinary'
+  },
+  defaults: {
+    cloudName: process.env.CLOUDINARY_CLOUD_NAME,
+  },
+  setup (options, nuxt) {
+    const resolver = createResolver(import.meta.url)
+
+    if (!options.cloudName) throw new Error('`@nuxtjs/cloudinary`: Property `cloudName` or variable `CLOUDINARY_CLOUD_NAME` missing. ')
```

Comment:
Hey @huang-julien 

By default it should be set in the config but user can also set it by using the useCldImageUrl composable as a part of the config object :)

Reply:
Nice ! but maybe can we remove line 20 (or set "" by default) ? 
Some users loads all their configs at server startup time :thinking: , having this would probably force users to set an empty string

Reply:
Ok sure, I will remove it then :)